docs/index.md

```diff
@@ -86,24 +86,87 @@ connection "terraform" {
+  # Defaults to CWD
+  paths = [ "*.tf" ]
+}
+```
 
-  # For example:
-  #  - "*.tf" matches all Terraform configuration files in the CWD
-  #  - "**/*.tf" matches all Terraform configuration files in the CWD and all sub-directories
-  #  - "../*.tf" matches all Terraform configuration files in the CWD's parent directory
-  #  - "steampipe*.tf" matches all Terraform configuration files starting with "steampipe" in the CWD
-  #  - "/path/to/dir/*.tf" matches all Terraform configuration files in a specific directory
-  #  - "/path/to/dir/main.tf" matches a specific file
+### Setting up paths
 
-  # If paths includes "*", all files (including non-Terraform configuration files) in
-  # the CWD will be matched, which may cause errors if incompatible file types exist
+The argument `paths` in the config is a list of directory paths, a GitHub repository URL, or a S3 URL to search for Terraform files. Paths may [include wildcards](https://pkg.go.dev/path/filepath#Match) and also support `**` for recursive matching. Defaults to the current working directory. For example:
 
-  # Defaults to CWD
-  paths = [ "*.tf" ]
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [
+    "*.tf",
+    "~/*.tf",
+    "github.com/turbot/polygoat//*.tf",
+    "github.com/turbot/polygoat//testing_frameworks/steampipe_mod_benchmark//*.tf",
+    "git::https://github.com/turbot/steampipe-plugin-alicloud.git//alicloud-test/tests/alicloud_account//*.tf",
+    "s3::https://s3.amazonaws.com/bucket/terraform_examples//**/*.tf"
+  ]
+}
+```
+
+#### Configuring local file paths
+
+You can define a list of local directory paths to search for terraform files. Paths are resolved relative to the current working directory. For example:
+
+- `*.tf` matches all Terraform configuration files in the CWD.
+- `**/*.tf` matches all Terraform configuration files in the CWD and all sub-directories.
+- `../*.tf` matches all Terraform configuration files in the CWD's parent directory.
+- `steampipe*.tf` matches all Terraform configuration files starting with "steampipe" in the CWD.
+- `/path/to/dir/*.tf` matches all Terraform configuration files in a specific directory. For example:
+  - `~/*.tf` matches all Terraform configuration files in the home directory.
+  - `~/**/*.tf` matches all Terraform configuration files recursively in the home directory.
+- `/path/to/dir/main.tf` matches a specific file.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "*.tf", "~/*.tf", "/path/to/dir/main.tf" ]
+}
+```
+
+**NOTE:** If paths includes `*`, all files (including non-Terraform configuration files) in the CWD will be matched, which may cause errors if incompatible file types exist.
+
+#### Configuring GitHub URLs
+
+You can define a list of URL as input to search for terraform files from a variety of protocols. For example:
+
+- `github.com/turbot/polygoat//*.tf` matches all top-level Terraform configuration files in the specified github repository.
```

Comment:
I tried to use this in my paths, but I was prompted for my GitHub login:
```
cbruno@M1P ~ % steampipe query
Welcome to Steampipe v0.17.0-alpha.28
For more information, type .help
Username for 'https://github.com':
```
If I try to ctrl+c without entering my username, I go into a state of limbo and have to kill Steampipe from another terminal:
```
cbruno@M1P ~ % vim ~/.steampipe/config/terraform.spc
cbruno@M1P ~ % steampipe query
Welcome to Steampipe v0.17.0-alpha.28
For more information, type .help
                                 >
>
> .quit


.quit
```

Right after I kill the Steampipe process and tried again, I got the error below and Steampipe crashed:
```
bruno@M1P ~ % steampipe query
Welcome to Steampipe v0.17.0-alpha.28
For more information, type .help

Error: connection closed before server preface received
```

On my 3rd try, when I tried to enter my username into the GitHub username prompt and hit enter, it seemed to break:
```
cbruno@M1P ~ % steampipe query
Welcome to Steampipe v0.17.0-alpha.28
For more information, type .help
                                 > cbruno10
⠹ Initializing...
```

So I have 2 questions from this interaction:
- Is logging into Git required? Is there a step users have to run first? I have Git configured on my system already and typically use SSH to pull/push with repos.
- It seems like the Steampipe CLI  doesn't expect or handle this prompt gracefully today, did you run into this at all while testing?

I'll also note that using the `git::...` example below, I was not prompted and I was able to query the correct tables + rows.

Reply:
Another note, it prompts me for my GitHub username even before running any queries (and what looks like to be before the `Initializing...` step finishes).